dds_web/security/auth.py

```diff
@@ -84,6 +84,18 @@ def verify_token_no_data(token):
     return user
 
 
+def verify_password_reset_token(token):
+    claims = __verify_general_token(token)
+    user = __user_from_subject(claims.get("sub"))
+    if user:
+        rst = claims.get("rst")
+        del claims
+        gc.collect()
+        if rst and rst == "pwd":
+            return user
+    raise AuthenticationError(message="Invalid token")
+
```

Comment:
If you have this function, then you have to update verify token with the following. Otherwise someone can use the system with a reset token.

```
@auth.verify_token
def verify_token(token):
    """Verify token used in token authentication."""
    claims = __verify_general_token(token=token)

    if claims.get("rst"):
        raise AuthenticationError(message="Invalid token")

    user = __user_from_subject(subject=claims.get("sub"))

    return __handle_multi_factor_authentication(
        user=user, mfa_auth_time_string=claims.get("mfa_auth_time")
    )
```

Reply:
Yes of course! I'll add a test for it as well.

Reply:
Fixed!